File: Patrones/Builder/Java/Builder/src/example/Pizza.java
package example;

/* "Product" */
public class Pizza {
	private String dough = "";
	private String sauce = "";
	private String topping = "";

	public String getDough(){
		return this.dough;
	}
	
	public String getSauce(){
		return this.sauce;
	}
	
	public String getTopping(){
		return this.topping;
	}
	
	public void setDough(String dough){
		this.dough = dough; 
	}
	
	public void setSauce(String sauce){
		this.sauce = sauce;
	}
	
	public void setTopping(String topping){
		this.topping = topping;
	}
}

File: Patrones/Builder/C++/src/Pizza.cpp
#include <Pizza.h>

string Pizza::getSauce(){
	return this->sauce;
}

string Pizza::getDough(){
	return this->dough;
}

string Pizza::getTopping(){
	return this->topping;
}

void Pizza::setDough(string dough){
	this->dough = dough; 
}
	
void Pizza::setSauce(string sauce){
	this->sauce = sauce;
}

void Pizza::setTopping(string topping){
	this->topping = topping;
}

Pizza::~Pizza(){}